src/workerd/api/performance.c++

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "performance.h"
+
+#include <workerd/io/io-util.h>
+
+#include <kj/encoding.h>
+
+namespace workerd::api {
+
+double Performance::now() {
+  // We define performance.now() for compatibility purposes, but due to Spectre concerns it
+  // returns exactly what Date.now() returns.
+  return dateNow();
+}
+
+v8::Local<v8::Value> PerformanceEntry::toJSON(jsg::Lock& js) {
+  auto result = js.obj();
+  result.set(js, "name"_kj, js.str(name));
+  result.set(js, "entryType"_kj, js.str(entryType));
+  result.set(js, "startTime"_kj, js.num(startTime));
+  result.set(js, "duration"_kj, js.num(duration));
+  return result;
```

Comment:
Likewise here, I would just throw. Users can't create these and we aren't either, there's no point to having an implementation here.